### PR TITLE
Common: Fix BitField mask generation.

### PR DESCRIPTION
**Pull request by lioncash**
See the [citra pull request](https://github.com/citra-emu/citra/pull/914) this is brought over from for more information.

### CODE REVIEW

**Comment by neobrain** [comment]
Unit tests? :(

**Comment by lioncash** [comment]
Sure thing, will update the PR when I have time.

**Comment by Tilka** [comment]
Ping.

File: Source/Core/Common/BitField.h [hunk 1]
```diff
@@ -183,7 +183,7 @@ struct BitField
 
 	__forceinline StorageType GetMask() const
 	{
-		return ((~(StorageTypeU)0) >> (8 * sizeof(T) - bits)) << position;
+		return (((StorageTypeU)~0) >> (8 * sizeof(T) - bits)) << position;
```
Comment (hrydgard) [inline Source/Core/Common/BitField.h]: Seems to me that this would break if StorageType is 64-bit, as ~0 is a 32-bit value.

Comment (Tilka) [inline Source/Core/Common/BitField.h]: ~0 will be sign-extended.

Comment (Tilka) [inline Source/Core/Common/BitField.h]: Then again, I don't understand the purpose of this commit.

**Comment by neobrain** [comment]
So uhm. Do we have unit tests, yet?

**Pull request merged**